src/lib.rs

```diff
@@ -144,43 +144,49 @@ pub fn rebuild(stack: &mut Vec<Node>, proof: &Multiproof) -> Node {
                     children[*digit] = node;
                     stack.push(FullNode(children))
                 } else {
-                    panic!("Could not pop a value from the stack, that is required for a BRANCH")
+                    return Err(format!(
+                        "Could not pop a value from the stack, that is required for a BRANCH"
+                    ));
                 }
             }
             EXTENSION(key) => {
                 if let Some(node) = stack.pop() {
                     stack.push(Extension(key.to_vec(), Box::new(node)));
                 } else {
-                    panic!("Could not find a node on the stack, that is required for an EXTENSION")
+                    return Err(format!(
+                        "Could not find a node on the stack, that is required for an EXTENSION"
+                    ));
                 }
             }
             ADD(digit) => {
                 if let (Some(el1), Some(el2)) = (stack.pop(), stack.last_mut()) {
                     match el2 {
                         FullNode(ref mut n2) => {
                             if *digit >= n2.len() {
-                                panic!(format!(
+                                return Err(format!(
                                     "Incorrect full node index: {} > {}",
                                     digit,
                                     n2.len() - 1
-                                ))
+                                ));
                             }
 
                             // A hash needs to be fed into the hash sponge, any other node is simply
                             // a child (el1) of the parent node (el2). this is done during resolve.
                             n2[*digit] = el1;
                         }
-                        Hash(_, _) => panic!("Hash node no longer supported in this case"),
-                        _ => panic!("Unexpected node type"),
+                        Hash(_, _) => {
+                            return Err(format!("Hash node no longer supported in this case"))
+                        }
+                        _ => return Err(format!("Unexpected node type")),
                     }
                 } else {
-                    panic!("Could not find enough parameters to ADD")
+                    return Err(format!("Could not find enough parameters to ADD"));
                 }
             }
         }
     }
 
-    stack.pop().unwrap()
```

Comment:
I would prefer that we called `map_err` to convert from any error to `String`, instead of unwrapping the error and then re-wrapping it

Reply:
Used `ok_or` which converts an `Option` into a `Result`